### PR TITLE
point-get: calculate the row level checksum on the real time to avoid incorrect checksum caused by schema change

### LINKED ISSUE #52590: tidb_row_checksum() calculate the returned value on the real time instead of previous stored value
## Enhancement

At the moment, when use `tidb_row_checksum()` function return the checksum value which was calculated when the data inserted, this value may be stable since the table schema change.

```
create table t (a int primary key, b int);
insert into t values (1, 2);                         # assume the checksum is `123`

alter table t add column c int default 1; 

select tidb_row_checksum() from t where a = 1;  ## `123` is returned, but the real checksum may be `234`
```

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### What problem does this PR solve?

Issue Number: close #52590 

Problem Summary:

### What changed and how does it work?

* `tidb_row_checksum` calculate the column-level checksum in the real time, instead of return the checksum store in the TiKV when the row inserted

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/executor/point_get.go [hunk 1]
```diff
@@ -400,6 +407,93 @@ func (e *PointGetExecutor) Next(ctx context.Context, req *chunk.Chunk) error {
...
+) error {
+	if !rowcodec.IsNewFormat(val) {
+		return nil
+	}
+	targetIndex, ok := shouldFillRowChecksum(schema)
+	if !ok {
+		return nil
+	}
+
+	columns := tblInfo.Cols()
```
[the remark this answers is not in the document]
Comment (3AceShowHand) [reply pkg/executor/point_get.go:431]: I think the checksum should be calculated on all columns shown to the user on the client query they query the row.

[the remark this answers is not in the document]
Comment (3AceShowHand) [reply pkg/executor/point_get.go]: the checksum should be calculated on all columns, decoded from the underline raw bytes.

**Review by zyguan — APPROVED** [review]

File: pkg/executor/point_get.go [hunk 2]
```diff
@@ -387,19 +389,122 @@ func (e *PointGetExecutor) Next(ctx context.Context, req *chunk.Chunk) error {
...
+	}
+	tz := sctx.GetSessionVars().TimeZone
+	ft := []*types.FieldType{schema.Columns[checksumColumnIndex].GetType()}
+	checksumCols := chunk.NewChunkWithCapacity(ft, req.Capacity())
+	for i := start; i < end; i++ {
+		handle, val := handles[i], values[i]
+		if !rowcodec.IsNewFormat(val) {
+			checksumCols.AppendNull(0)
+			continue
+		}
```
Comment (tangenta) [inline pkg/executor/point_get.go:459]: Will this be explained in user documentation?

Comment (3AceShowHand) [reply pkg/executor/point_get.go:459]: no, there is no changes compare to the previous implementation, just fill the null as placeholder, and reset by the `fillChecksum` method.

Comment (3AceShowHand) [reply pkg/executor/point_get.go:459]: I got your point, I think we can update the documentation to let user's know tat checksum only calculated on the new format.

**Review by tangenta — APPROVED** [review]

**Review by XuHuaiyu — APPROVED** [review]

**Pull request merged**